### ISSUE TITLE
jiralert returns HTTP 500 when jira user does not have access to project

### ISSUE THREAD

**Issue by nvtkaszpir** [prometheus-community/jiralert#53]
Looks like jiralert returns HTTP 500 when it tries to access remote jira instance and uses good credentials but the given user does not have access to the given jira project.
Also it helps to enable json log format to catch the error properly, logfmt is sometimes producing not very useful error messages.

example error:
err:
```
JIRA request https://REDACTED.atlassian.net/rest/api/2/search?jql=project%3D%22XX%22+and+labels%3D%22ALERT%7Balertname%3D%5C%22KubeJobFailed%5C%22%2Ccluster_name%3D%5C%22REDACTED...+order+by+resolutiondate+desc&startAt=0&maxResults=2&expand=&fields=summary,status,resolution,resolutiondate&validateQuery= returned status 403 Forbidden, body ""
```
msg: ```error handling request```
statusCode: 500
statusText: ```Internal Server Error```

I think jiralert could pass back 403 error from jira?

**Comment by nvtkaszpir** [comment]
similiar issue was reported in #35

**Comment by free** [comment]
You are correct, JIRAlert is [setting the `retry` flag](https://github.com/prometheus-community/jiralert/blob/7a1b0b33f457efc9b7f8b4d36bb031b7fe0f68dc/pkg/notify/notify.go#L241) to true if Jira's response code is 500 or 503, false otherwise. Then, if the `retry` flag is true [JIRAlert responds](https://github.com/prometheus-community/jiralert/blob/7a1b0b33f457efc9b7f8b4d36bb031b7fe0f68dc/cmd/jiralert/main.go#L99) to Alertmanager with a 503 Service Unavailable, else with a 500 Internal Server Error.

I guess it could respond with the original status code, but would that make any difference WRT Alertmanager's behavior? Currently the intent is to simply have Alertmanager retry the webhook as long as we respond with a 503 (i.e. if Jira responded with 500 or 503) and not retry it otherwise.

But looking at the Alertmanager code, it looks like it will [retry all 5xx codes](https://github.com/prometheus/alertmanager/blob/655947d7e0ea0a15815a6348fcf74a7ca79f8e91/notify/util.go#L175) for webhooks ([and only those](https://github.com/prometheus/alertmanager/blob/94d875f1227b29abece661db1a68c001122d1da5/notify/webhook/webhook.go#L59)) so it may make sense to either pass through the original response code or simply respond with something other than 500 Internal Server Error if JIRAlert decides it is not a retriable error.

**Comment by SuperQ** [comment]
**This project has been deprecated in favor of [direct alertmanager integration](https://prometheus.io/docs/alerting/latest/configuration/#jira_config).**
